src/include/daos_srv/pool.h

```diff
@@ -22,6 +22,48 @@
 #include <daos_pool.h>
 #include <daos_security.h>
 #include <gurt/telemetry_common.h>
+#include <daos_srv/policy.h>
+
+/** Metrics for each individual active pool */
+struct ds_pool_metrics {
+	uuid_t			pm_pool_uuid;
+	ABT_mutex		pm_lock; /* multiple threads may have access */
+
+	struct d_tm_node_t	*pm_started_timestamp;
+	/* TODO: add more per-pool metrics */
+};
+
+/**
+ * Lock a pool metrics structure for synchronized access.
+ *
+ * \param[in]	metrics		Pool metrics
+ */
+static inline void
+ds_pool_metrics_lock(struct ds_pool_metrics *metrics)
+{
+	if (unlikely(metrics == NULL))
+		return;
+
+	ABT_mutex_lock(metrics->pm_lock);
+}
+
+/**
+ * Unlock a pool metrics structure for synchronized access.
+ *
+ * \param[in]	metrics		Pool metrics
+ */
+static inline void
+ds_pool_metrics_unlock(struct ds_pool_metrics *metrics)
+{
+	if (unlikely(metrics == NULL))
+		return;
+
+	ABT_mutex_unlock(metrics->pm_lock);
+}
+
+struct ds_pool_metrics *ds_pool_metrics_get(const uuid_t pool_uuid);
+int ds_pool_metrics_get_path(const uuid_t pool_uuid, char *path,
+			     size_t path_len);
```

Comment:
I don't quite see what are these changes for? They should be removed if they are not used in this patch.

Reply:
Probably a merge conflict. @kmajzero could you please remove this when you refresh the patch to address the CI failure please?